### ISSUE TITLE
[V3 Filter] Unable to filter sentences

### ISSUE THREAD

**Issue by Crossedfall** [Cog-Creators/Red-DiscordBot#1904]
# Command bugs

#### Command name

`filter add "<sentence>"`

#### What cog is this command from?

Filter [core]

#### What were you expecting to happen?

Whenever the full sentence is posted within the filtered Discord server, it should be removed by the bot.

#### What actually happened?

Nothing. Sentences will be deleted if you remove the spaces from the trigger phrase, however.

#### How can we reproduce this issue?

1. Create a new filter with spaces
2. Attempt to trigger the filter

![filter gif](https://user-images.githubusercontent.com/26130695/42119867-c0200498-7bd6-11e8-9c8e-689eb104a67a.gif)

_Note: it would seem that some sentences are read whereas most are not. Multiple tests are advised_

#### Red Version
![image](https://user-images.githubusercontent.com/26130695/42119883-e7749ff4-7bd6-11e8-9b0e-75283ea3d080.png)

**Comment by mikeshardmind** [comment]
Unfortunately, that gif isn't a demonstration of anything not working.

The filter will never impact other bots, the bot owner, mods or admins:

https://github.com/Cog-Creators/Red-DiscordBot/blob/3bcf375204c22271ad3ed1fc059b598b751aa03f/redbot/cogs/filter/filter.py#L248-L251

https://github.com/Cog-Creators/Red-DiscordBot/blob/3bcf375204c22271ad3ed1fc059b598b751aa03f/redbot/core/utils/mod.py#L143-L150

if you need to test filter's behavior, if you launch the bot with `--dev` you can use `[p]mockmsg` or you can get someone else to help you test it.

**Comment by Crossedfall** [comment]
@mikeshardmind if you watch the gif to the end you'll note that it filters the message if I remove all of the spaces. That'd imply that it does actually filter messages from a bot owner and guild admin.

If it really isn't supposed to do that, then there might be another issue at play, possibly unrelated to this report. 

Edit: I have tested this with other users and in other servers. I can confirm that even without this oddity it still happens.

**Comment by mikeshardmind** [comment]
Does the audit log confirm it was your bot, and are you running more than one bot on that token?

If the answer to those is "yes" and "no" respectively, something is seriously wrong here.

**Comment by Crossedfall** [comment]
As far as I can tell, filtering words is not tracked in the log; even with the `--debug` flag. No other bots are running with this token.

I have done a number of tests on this and I can confirm without a shadow of a doubt that it will filter away my messages without a problem. Have you tested and had different results?

**Comment by mikeshardmind** [comment]
Well, something is seriously wrong.... I can confirm both the issue with the spacing and with the bot owner being considered for deletion.

**Comment by Crossedfall** [comment]
@mikeshardmind **Update:** In B17, I can confirm that the bot no longer removes owner messages. The filter is still broken unfortunately.

**Comment by palmtree5** [comment]
oh wait, I know exactly why bot owner messages are being considered for deletion

```py
if user and user == await bot.is_owner(user): 
     return True
```

`bot.is_owner` returns a bool. Hence why bot owner messages are being considered for deletion. That's actually a really easy fix though as you really just need to drop `user == ` from that.

The sentences issue results from the way it's added to the filter. Consider the following:

https://github.com/Cog-Creators/Red-DiscordBot/blob/V3/develop/redbot/cogs/filter/filter.py#L76-L87

Notice under the outermost `else` in the `for` loop here that no spaces are ever actually added back in when adding words in the sentence to `tmp`.

I'm gonna assign this to myself to do as both are simple fixes

**Comment by Tobotimus** [comment]
Resolved by #2065.
